tests/player.c

```diff
@@ -536,6 +439,335 @@ START_TEST (test_play_audio_eos)
+    TestPlayerState * old_state, TestPlayerState * new_state)
+{
+  if (change == STATE_CHANGE_MEDIA_INFO_UPDATED)
+    test_media_info_object (player, new_state->media_info);
+  else if (change == STATE_CHANGE_END_OF_STREAM)
+    g_main_loop_quit (new_state->loop);
+}
+
+START_TEST (test_play_media_info)
+{
+  GstPlayer *player;
+  TestPlayerState state;
+  gchar *uri;
+
+  memset (&state, 0, sizeof (state));
+  state.loop = g_main_loop_new (NULL, FALSE);
+  state.test_callback = test_play_media_info_cb;
+
+  player = test_player_new (&state);
+
+  fail_unless (player != NULL);
+
+  uri = gst_filename_to_uri (TEST_PATH "/sintel.mkv", NULL);
+  fail_unless (uri != NULL);
+  gst_player_set_uri (player, uri);
+  g_free (uri);
+
+  gst_player_play (player);
+  g_main_loop_run (state.loop);
+
+  g_object_unref (player);
+  g_main_loop_unref (state.loop);
+}
+
+END_TEST;
+
+static void
+switch_streams (GstPlayer * player, GType type)
+{
+  GList *list = NULL;
+  gint current_index = -1;
+  GstPlayerMediaInfo *media_info;
+
+  media_info = gst_player_get_media_info (player);
+  fail_unless (media_info != NULL);
+
+  if (type == GST_TYPE_PLAYER_SUBTITLE_INFO) {
+    GstPlayerSubtitleInfo *sub_info;
+
+    sub_info = gst_player_get_current_subtitle_track (player);
+    fail_unless (sub_info != NULL);
+    current_index = gst_player_stream_info_get_index
+                    ((GstPlayerStreamInfo *) sub_info);
+    list = gst_player_get_subtitle_streams (media_info);
+  } else if (type == GST_TYPE_PLAYER_AUDIO_INFO) {
+    GstPlayerAudioInfo *audio_info;
+
+    audio_info = gst_player_get_current_audio_track (player);
+    fail_unless (audio_info != NULL);
+    current_index = gst_player_stream_info_get_index
+                    ((GstPlayerStreamInfo *) audio_info);
+    list = gst_player_get_audio_streams (media_info);
+  }
+
+  for (; list != NULL; list = list->next) {
+    gint index;
+    GstPlayerStreamInfo *stream = (GstPlayerStreamInfo *) list->data;
+
+    index = gst_player_stream_info_get_index (stream);
+    if (index == current_index)
+      continue;
+
+    {
+      GstPlayerStreamInfo *new_stream = NULL;
+
+      /* set the new track and compare it with index returned from the player */
+      if (type == GST_TYPE_PLAYER_SUBTITLE_INFO) {
+        fail_unless (gst_player_set_subtitle_track (player, index) == TRUE);
+        new_stream = (GstPlayerStreamInfo *)
+                          gst_player_get_current_subtitle_track (player);
+      } else if (type == GST_TYPE_PLAYER_AUDIO_INFO) {
+        fail_unless (gst_player_set_audio_track (player, index) == TRUE);
+        new_stream = (GstPlayerStreamInfo *)
+                          gst_player_get_current_audio_track (player);
+      }
+
+      fail_unless_equals_int (gst_player_stream_info_get_index (new_stream),
+                              index);
+      /* wait for 2 seconds before switching to next stream */
+      sleep (2);
```

Comment:
one question, the reason behind sleep was that I was planning to do something like this in test:

- save the current player position (cur_position)
- switch to new stream 
- sleep for 3 seconds 
- query the current position again 
- compare the new current position with previous position and if the position is not changed then 
pipeline is stalled somewhere.

Is it okay to call sleep() in this case ?  I have ran into instances where the stream is switched but then it get stalled during playback. If we simply query the current stream index then it will give us wrong data. Of course,  if you wait for 2 min then tcase will timeout and that will trigger failure ;) 

Let me know your thought